src/main/kotlin/com/exactpro/th2/ws/client/ControlService.kt

```diff
@@ -0,0 +1,67 @@
+package com.exactpro.th2.ws.client
+
+import com.exactpro.th2.conn.grpc.ConnGrpc.ConnImplBase
+import com.exactpro.th2.conn.grpc.Response
+import com.exactpro.th2.conn.grpc.Response.Status.FAILURE
+import com.exactpro.th2.conn.grpc.Response.Status.SUCCESS
+import com.exactpro.th2.conn.grpc.StartRequest
+import com.exactpro.th2.conn.grpc.StopRequest
+import io.grpc.stub.StreamObserver
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.Executors
+import java.util.concurrent.Future
+import java.util.concurrent.TimeUnit.SECONDS
+
+class ControlService(
+    startClient: () -> Unit,
+    stopClient: () -> Unit
+) : ConnImplBase(), AutoCloseable {
+    @Volatile private var isRunning = false
+    private val start = { startClient(); isRunning = true }
+    private val stop = { stopClient(); isRunning = false }
+    private val executor = Executors.newSingleThreadScheduledExecutor()
+    private var stopFuture: Future<*> = CompletableFuture.completedFuture(null)
+
+    override fun start(request: StartRequest, observer: StreamObserver<Response>) = observer {
+        when {
+            isRunning -> failure("Already running")
+            !stopFuture.isDone -> failure("Scheduled stop has not yet completed")
+            else -> {
+                start()
+                success("Successfully started")
+            }
+        }
+    }
+
+    override fun stop(request: StopRequest, observer: StreamObserver<Response>) = observer {
+        when {
+            !isRunning -> failure("Already stopped")
+            !stopFuture.isDone -> failure("Stop is already scheduled")
+            request.after > 0 -> {
+                stopFuture = executor.schedule(stop, request.after.toLong(), SECONDS)
```

Comment:
These steps should be execute atomically because more than one request can be handle at the same time